File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ChannelRenameEvent.java
package work.socialhub.kslack.entity.event;

/**
 * The channel_rename event is sent to all connections for a workspace when a channel is renamed.
 * Clients can use this to update their local list of channels.
 * <p>
 * https://api.slack.com/events/channel_rename
 */
public class ChannelRenameEvent implements Event {

    public static final String TYPE_NAME = "channel_rename";

    private final String type = TYPE_NAME;
    private Channel channel;

    public String getType() {
        return this.type;
    }

    public Channel getChannel() {
        return this.channel;
    }

    public void setChannel(Channel channel) {
        this.channel = channel;
    }

    public static class Channel {
        private String id;
        private String name; // new name
        private Integer created;

        public String getId() {
            return this.id;
        }

        public String getName() {
            return this.name;
        }

        public Integer getCreated() {
            return this.created;
        }

        public void setId(String id) {
            this.id = id;
        }

        public void setName(String name) {
            this.name = name;
        }

        public void setCreated(Integer created) {
            this.created = created;
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ChannelRenameEvent.kt
package work.socialhub.kslack.entity.event

import kotlinx.serialization.Serializable

/**
 * The channel_rename event is sent to all connections for a workspace when a channel is renamed.
 * Clients can use this to update their local list of channels.
 *
 *
 * https://api.slack.com/events/channel_rename
 */
@Serializable
class ChannelRenameEvent : Event {
    override val type: String = TYPE_NAME
    var channel: Channel? = null

    @Serializable
    class Channel {
        var id: String? = null
        var name: String? = null // new name
        var created: Int? = null
    }

    companion object {
        const val TYPE_NAME: String = "channel_rename"
    }
}